common/statistics/src/clients/packet_statistics.rs

```diff
@@ -1,48 +1,21 @@
+use super::ClientStatsEvents;
 use std::{
     collections::VecDeque,
     time::{Duration, Instant},
 };
 
 use nym_metrics::{inc, inc_by};
+use serde::{Deserialize, Serialize};
 use si_scale::helpers::bibytes2;
 
-// Metrics server
-use futures::future::{FusedFuture, OptionFuture};
-use futures::FutureExt;
-#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
-use http_body_util::Full;
-#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
-use hyper::body::Bytes;
-#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
-use hyper::server::conn::http1;
-#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
-use hyper::service::service_fn;
-#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
-use hyper::{Request, Response};
-#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
-use hyper_util::rt::TokioIo;
-#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
-use std::convert::Infallible;
-#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
-#[cfg(feature = "metrics-server")]
-use std::net::SocketAddr;
-#[cfg(not(all(target_arch = "wasm32", target_os = "unknown")))]
-use tokio::net::TcpListener;
-
-use crate::spawn_future;
-
-// Time interval between reporting packet statistics
-const PACKET_REPORT_INTERVAL_SECS: u64 = 2;
-// Interval for taking snapshots of the packet statistics
-const SNAPSHOT_INTERVAL_MS: u64 = 500;
 // When computing rates, we include snapshots that are up to this old. We set it to some odd number
 // a tad larger than an integer number of snapshot intervals, so that we don't have to worry about
 // threshold effects.
 // Also, set it larger than the packet report interval so that we don't miss notable singular events
 const RECORDING_WINDOW_MS: u64 = 2300;
```

Comment:
if we're going to be moving configuration values to config, can we move this one too?

Reply:
Yep, will do in the config PR next

Reply:
I notice that a set of variables that are deeply tied are split into different locations.
These values are judiciously set for how it affects the UI presentation in the vpn client. It feels like these now possibly start meaning different things now that they also used for other things?

Reply:
With the new collection module, the control is moved elsewhere. I made sure to keep that same small reporting interval to keep that module's behavior identical, and so other module can also profit from that